src/Draco.Compiler/Internal/Syntax/Parser.cs

```diff
@@ -1404,6 +1411,21 @@ private bool CanBailOut(SyntaxNode node)
         return node.LastToken?.TrailingTrivia.Any(t => t.Kind == TriviaKind.Newline) == true;
     }
 
+    /// <summary>
+    /// Checks a <see cref="SyntaxToken"/> for whether it is a heritage token and reports an error in case it is.
+    /// </summary>
+    /// <param name="token">The token to check.</param>
+    /// <param name="syntaxKind">The text which is displayed in the reported diagnostic indicating what kind of syntactic element the heritage token is.</param>
+    private void CheckHeritageToken(SyntaxToken token, string syntaxKind)
+    {
+        if (SyntaxFacts.GetHeritageReplacement(token.Kind) is { } replacementKind)
```

Comment:
You can invert this condition and return to make it a guard statement, less indentation, less happy-path complexity

Reply:
oooh right, forgot since I just copy-pasted the if statement